ocaml/database/db_cache_types.ml

```diff
@@ -16,6 +16,25 @@ open Db_exn
 
 module Time = struct type t = Generation.t end
 
+module HashedString = struct
+  type t = string
+
+  let equal = String.equal
+
+  let hash = Hashtbl.hash
+end
+
+module StringPool = Weak.Make (HashedString)
+
+let pool = StringPool.create 2048
+
+let unique str =
+  match StringPool.find_opt pool str with
```

Comment:
I think `StringPool.merge pool str` would already do this. The docs say "merge t x returns an instance of x found in t if any, or else adds x to t and return x."